tests/test_spstats.py

```diff
@@ -156,6 +156,46 @@ def test_neff_estimation(self):
 
                 assert np.abs(neff_circ_exact-neff_circ_numer) < 0.001
 
+class TestSubSampling:
+
+    def test_circular_masking(self):
+        """Test that the circular masking works as intended"""
+
+        # using default (center should be [2,2], radius 2)
+        circ = xdem.spstats.create_circular_mask(5,5)
+        circ2 = xdem.spstats.create_circular_mask(5,5,center=[2,2],radius=2)
+
+        assert np.array_equal(circ,circ2)
+
+        # check distance is not a multiple of pixels (more accurate subsampling)
+
+        # will create a 1-pixel mask around the center
+        circ3 = xdem.spstats.create_circular_mask(5,5,center=[1,1],radius=1)
+        # will create a square mask (<1.5 pixel) around the center
+        circ4 = xdem.spstats.create_circular_mask(5,5,center=[1,1],radius=1.5)
+
+        assert not np.array_equal(circ3,circ4)
+
+
+    def test_ring_masking(self):
+        """Test that the ring masking works as intended"""
+
+        # by default, the mask is only False (ring of size 0)
+        ring1 = xdem.spstats.create_ring_mask(5,5)
+
+        assert np.array_equal(ring1,np.zeros((5,5)))
+
+        # test rings with different inner radius
```

Comment:
Same comment, you could create the matrix as it should look like and compare that they are the same!

Reply:
Added